src/main/java/core/basesyntax/strategy/impl/BalanceHandlerImpl.java

```diff
@@ -1,15 +1,22 @@
 package core.basesyntax.strategy.impl;
 
 import core.basesyntax.dao.FruitDao;
-import core.basesyntax.dao.FruitDaoImpl;
 import core.basesyntax.model.Transaction;
 import core.basesyntax.strategy.OperationHandler;
 
 public class BalanceHandlerImpl implements OperationHandler {
-    private final FruitDao fruitDao = new FruitDaoImpl();
+    private final FruitDao fruitDao;
+
+    public BalanceHandlerImpl(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
 
     @Override
     public void applyChanges(Transaction transaction) {
-        fruitDao.add(transaction.getFruit(), transaction.getAmountFruits());
+        int remaining =
+                fruitDao.getQuantity(transaction.getNameFruit()) == null
+                ? 0 : fruitDao.getQuantity(transaction.getNameFruit());
```

Comment:
You don't use this check in other handlers. Maybe checking fruitName for null will be better while Transaction initializing?